carma_wm_ctrl/src/GeofenceSchedule.cpp

```diff
@@ -14,92 +14,111 @@
  * the License.
  */
 
-#include <ros/time.h>
-#include <carma_wm_ctrl/GeofenceSchedule.h>
+#include <carma_wm_ctrl/GeofenceSchedule.hpp>
 #include <boost/date_time/posix_time/posix_time.hpp>
-#include <ros/ros.h>
 namespace carma_wm_ctrl
 {
+
+double toSec(const boost::posix_time::time_duration& duration) {
+  if (duration.is_special()) {
+    throw std::invalid_argument("Cannot convert special duration to seconds");
+  }
+  return duration.total_microseconds() / 1000000.0;
+}
+
+double toSec(const boost::posix_time::ptime& time) {
+  return toSec(time - boost::posix_time::from_time_t(0));
+}
+
 GeofenceSchedule::GeofenceSchedule()
 {
 }
 
-GeofenceSchedule::GeofenceSchedule(ros::Time schedule_start, ros::Time schedule_end, ros::Duration control_start,
-                                   ros::Duration control_duration, ros::Duration control_offset,
-                                   ros::Duration control_span, ros::Duration control_period, DayOfTheWeekSet week_day_set):
+GeofenceSchedule::GeofenceSchedule(rclcpp::Time schedule_start, rclcpp::Time schedule_end, rclcpp::Duration control_start,
+                                   rclcpp::Duration control_duration, rclcpp::Duration control_offset,
+                                   rclcpp::Duration control_span, rclcpp::Duration control_period, DayOfTheWeekSet week_day_set):
                                    schedule_start_(schedule_start), schedule_end_(schedule_end),
                                    control_start_ (control_start), control_duration_ (control_duration),
                                    control_offset_ (control_offset), control_span_ (control_span), control_period_ (control_period), week_day_set_(week_day_set)
 {}
 
-bool GeofenceSchedule::scheduleExpired(const ros::Time& time) const
+bool GeofenceSchedule::scheduleExpired(const rclcpp::Time& time) const
 {
+  
   return schedule_end_ < time;
 }
 
-bool GeofenceSchedule::scheduleStarted(const ros::Time& time) const
+bool GeofenceSchedule::scheduleStarted(const rclcpp::Time& time) const
 {
   return schedule_start_ <= time;
 }
 
-// returns ros::Time(0) when the schedule is expired or the next interval will be on a different day of the week
+// returns rclcpp::Time(0) when the schedule is expired or the next interval will be on a different day of the week
 // Argument provided as absolute time (since 1970)
-std::pair<bool, ros::Time> GeofenceSchedule::getNextInterval(const ros::Time& time) const
+std::pair<bool, rclcpp::Time> GeofenceSchedule::getNextInterval(const rclcpp::Time& time) const
 {
   if (scheduleExpired(time))
   {
-    ROS_DEBUG_STREAM("Geofence schedule expired");
-    return std::make_pair(false, ros::Time(0));  // If the schedule has expired or was never started
+    RCLCPP_DEBUG_STREAM(rclcpp::get_logger("carma_wm_ctrl::GeofenceSchedule"), "Geofence schedule expired");
```

Comment:
Something I realized recently is that this may not match to the values based in from the log file correctly unless only the package name is provided, so "carma_wm_ctrl" is the way to go for now even though its less descriptive.

Reply:
added to every logger